aspnet/fundamentals/dependency-injection.rst

```diff
@@ -1,10 +1,277 @@
+	The current set of services configured in the container. Available only to the ``ConfigureServices`` method, and used by that method to configure the services available to an application.
+
+Looking at each method in the ``Startup`` class in the order in which they are called, the following services may be requested as parameters:
+
+Startup Constructor
+	- ``IApplicationEnvironment``
+	- ``IHostingEnvironment``
+	- ``ILoggerFactory``
+	
+ConfigureServices
+	- ``IServiceCollection``
+
+Configure
+	- ``IApplicationBuilder``
+	- ``IApplicationEnvironment``
+	- ``IHostingEnvironment``
+	- ``ILoggerFactory``
+	
+
+.. note:: Although ``ILoggerFactory`` is available in the constructor, it is typically configured in the ``Configure`` method. Learn more about the ``Startup`` class constructor and ``Configure`` method in :doc:`startup`.
+
+The ``ConfigureServices`` method is responsible for defining the services the application will use, including platform features like Entity Framework and ASP.NET MVC. Initially, the ``IServiceCollection`` provided to ``ConfigureServices`` has just a handful of services defined. The default web template shows an example of how to add additional services to the container using a number of extensions methods like ``AddEntityFramework``, ``Configure<Options>``, and ``AddMvc``.
+
+.. literalinclude:: ../../common/samples/WebApplication1/src/WebApplication1/Startup.cs
+	:language: c#
+	:linenos:
+	:lines: 45-84
+	:dedent: 8
+	:emphasize-lines: 5,11,18,24
+
+In order to keep the ``ConfigureServices`` method manageable, it is recommended that :doc:`middleware` and :doc:`request-features` authors provide extension methods and fluent configuration APIs that work with ``IServiceCollection``. The built-in features for Entity Framework, Identity, and ASP.NET MVC all follow this approach and can be used as models for providing a very succinct way to add a great deal of functionality to an ASP.NET application.
+
+Of course, in addition to configuring the application to take advantage of various framework features, you can also use ``ConfigureServices`` to configure your own application services.
+
+Registering Your Own Services
+-----------------------------
+
+In the default web template example above, two application services are added to the ``IServiceCollection`` (``IEmailSender`` and ``ISmsSender``). The ``AddTransient`` method is used to map abstract types to concrete services that are instantiated separately for every object that requires it. This is known as the service's *lifetime*, and additional lifetime options are described below. It is important to choose an appropriate lifetime for each of the services you register. Should a new instance of the service be provided to each class that requests it? Should once instance be used throughout a given web request? Or should a single instance be used for the lifetime of the application?
```

Comment:
> default web template example above, two application services

Suggest highlighting the specific lines of the sample where these application services are added.

It's worth also mentioning that every bit of the `ConfigureServices()` method is about adding services. For example `services.AddMvc()` adds the services MVC requires and `services.Configure<T>()` adds or uses `IOptions<T>` and `IConfigureOptions<T>` services.